spark/src/main/scala/org/apache/spark/rdd/FilteredCartesianRDD.scala

```diff
@@ -62,9 +82,20 @@ sealed class FilteredCartesianRDD[T: ClassTag, U: ClassTag, V: ClassTag](
     val part2 = currSplit.s2 // partition for (|meta)rdd2
     require(metardd1.iterator(part1, context).hasNext, "Meta-RDD Failure")
     require(metardd2.iterator(part2, context).hasNext, "Meta-RDD Failure")
+
+    /* A call to RDD.iterator invokes the block manager to compute and fetch
+     * required partition of metardd1 and metardd2. Because subsequent calls
+     * to RDD.iterator are happening within the same TaskContext the partitions
+     * of metardds will only be computed once per task execution.
+     * This is essentially a partition-wise .collect it may cause network traffic.
+     */
```

Comment:
I think that you already addressed something that I wrote above with this comment.